lib/lita/adapters/slack/message_handler.rb

```diff
@@ -39,17 +39,29 @@ def handle
         attr_reader :type
 
         def body
-          normalized_message = if data["text"]
-            data["text"].sub(/^\s*<@#{robot_id}>/, "@#{robot.mention_name}")
+          normalized_text = nil
+          if data["text"]
+            normalized_text = data["text"].sub(/^\s*<@#{robot_id}>/, "@#{robot.mention_name}")
           end
 
-         normalized_message = remove_formatting(normalized_message) unless normalized_message.nil?
+          normalized_text = remove_formatting(normalized_text) unless normalized_text.nil?
 
-          attachment_text = Array(data["attachments"]).map do |attachment|
-            attachment["text"] || attachment["fallback"]
-          end
+          lines = Array(data["attachments"]).inject([normalized_text]){
+            |total, att| total.concat parse_attachment(att)
+          }
+          lines.compact.join("\n")
+        end
 
-          ([normalized_message] + attachment_text).compact.join("\n")
+        def parse_attachment(a)
```

Comment:
⛏ In a perfect world, this would be `attachement` rather than just `a`.

Reply:
It'll bug me now if I don't fix it - give me a sec